velox/exec/SortWindowBuild.cpp

```diff
@@ -249,6 +289,9 @@ void SortWindowBuild::noMoreInput() {
     // the partition. This will order the rows for getOutput().
     sortPartitions();
   }
+
+  // Releases the unused memory reservation after procesing input.
+  pool_->release();
```

Comment:
Can you also please keep this in dtor?

Reply:
kept, https://github.com/facebookincubator/velox/pull/11370/files#diff-8b2cfa9e2735214ab7f8f38f325c5584e52855cdfe5d88ad7329e571a1083580R39